### ISSUE TITLE
Branch.io + Spotlight

### ISSUE THREAD

**Issue by tnurov** [BranchMetrics/ios-branch-deep-linking#562]
Hello!

**First way**
I created more than 100 objects use this code

 ```
 BranchUniversalObject *object = [[BranchUniversalObject alloc] initWithCanonicalIdentifier:identifier];
    object.title = title;
    object.contentDescription = .contentDescription;
    object.keywords = keywords;
    object.imageUrl = object.imageUrl;
    
    [object addMetadataKey:@"url" value:urlString];


    [object listOnSpotlightWithCallback:^(NSString * _Nullable url, NSError * _Nullable error) {
        NSLog(@"url %@, error %@ ", url, error);
    }];
```

All callbacks was with url..
But I could find only a few objects in Spotlight.

**Second way**

I used new code for this objects

```
 BranchCSSearchableItemAttributeSet *set = [[BranchCSSearchableItemAttributeSet alloc] init];
    set.title = service.title;
    set.params = @{@"url" : urlString};
    set.keywords = [NSSet setWithArray:service.keywords];
    set.contentDescription = service.contentDescription;
    set.thumbnailURL = [NSURL URLWithString:[IMAGE_ROOT_URL_WITHOUT_SLASH stringByAppendingString:service.img]];
    
    [set indexWithCallback:^(NSString * _Nullable url, NSString * _Nullable spotlightIdentifier, NSError * _Nullable error) {
        NSLog(@"url %@, spotlightIdentifier %@, error %@ ", url, spotlightIdentifier, error);
    }];
```

All objects added in Spotlight. 

But 

1.
 If I sent my objects to Branch and closed app. Branch SDK saved queue. After reopen sent request and had response with branch-links. Object BranchSpotlightUrlRequest had empty spotlightCallback and Spotlight objects didn't create.

2.
I clicked in Spotlight on LINK-1. Branch-callback was with empty params.
I clicked in Spotlight on LINK-2. Branch-callback was with LINK-1 params.
I clicked in Spotlight on LINK-3. Branch-callback was with LINK-2 params.

[self getLatestReferringParams] returned previous params

**Comment by tnurov** [comment]
And 3.

I send  to Branch one object 3 times and I have 3 link in Spotlight

**Comment by E-B-Smith** [comment]
It looks like something fishy is going on.  Let me look into this and get back to you.

Thank you for the heads up.

**Comment by tnurov** [comment]
@E-B-Smith hello!

Yes, actions is very strange. I used Spotlight functional in other project with older version Branch SDK and it was absolutely simple.
Do you have result of research? 

Thank you for your help!

**Comment by hitendradeveloper** [comment]
First thing first,
I have integrated branch framework using Fabric app on 21st May 2017, drag and drop install.
I am sharing link using iMessage and opening the same from the iMessage for the test.

I am phasing the same issue, using the `BranchUniversalObject *object`
```
I clicked in Spotlight on LINK-1. Branch-callback was with empty params.
I clicked in Spotlight on LINK-2. Branch-callback was with LINK-1 params.
I clicked in Spotlight on LINK-3. Branch-callback was with LINK-2 params.

[self getLatestReferringParams] returned previous params
```

Also opening the url

**Comment by E-B-Smith** [comment]
This sounds like it's connected to an issue that Parth @parthkalavadia  and I are looking at now.
We'll update you all with what we find.

**Comment by parthkalavadia** [comment]
The issue about the spotlight could be resolved in SDK v0.2.12. Please update the SDK and try it again.

**Comment by parthkalavadia** [comment]
I am closing this issue as this is resolved.
